cirq/google/engine/engine.py

```diff
@@ -619,6 +619,103 @@ def remove_job_labels(self, job_resource_name: str, label_keys: List[str]):
             fingerprint = job.get('labelFingerprint', '')
             self._set_job_labels(job_resource_name, new_labels, fingerprint)
 
+    def list_processors(self, project_id: str) -> List[Dict]:
+        """ Returns a list of Processors that the user has visibility to in the
+            provided project. The names of these processors are used to identify
+            devices when scheduling jobs and gathering calibration metrics.
+
+        Params:
+            project_id: The ID of the Google Cloud project to check, e.g.
+                `my-project-123`
+
+        Returns:
+            A list of dictionaries containing the metadata of each processor.
```

Comment:
should we doc what is in these dictionaries?

Reply:
I'm concerned about this getting out of sync, but it does seem useful to get a sense of what to expect so I made a pass at what's returned today. PTAL.

Reply:
The service proto isn't published, so removing docs as discussed offline.